Test/ProductsTests.cs

```diff
@@ -49,15 +63,34 @@ public void AddAndRemoveProduct_CheckQuantityInCartAndButtonText()
             productsPage.ProductButton("Sauce Labs Fleece Jacket").Click();
             productsPage.ProductButton("Sauce Labs Fleece Jacket").Click();
 
-            string actualQuantityInCart = productsPage.CartLink().Text;
+            string actualQuantityInCart = headerPage.CartLink().Text;
             string actualButtonText = productsPage.ProductButton("Sauce Labs Fleece Jacket").Text;
             
             Assert.Multiple(() =>
                 {
-                    Assert.That(actualQuantityInCart, Is.EqualTo(expectedQuantitytInCart), $"Expected quantity in cart '{expectedQuantitytInCart}', but got '{actualQuantityInCart}'.");
-                    Assert.That(actualButtonText, Is.EqualTo(expectedButtonText), $"Expected the button text '{expectedButtonText}', but got '{actualButtonText}'.");
+                    Assert.That(actualQuantityInCart, Is.EqualTo(expectedQuantitytInCart));
+                    Assert.That(actualButtonText, Is.EqualTo(expectedButtonText));
                 }
             );
         }
+
+        [Test]
+        [AllureDescription("Test to check the operation of \"try-catch\" in Tear Down in case of a failed test")]
+        [AllureFeature("Tests to fail")]
+        public void CheckTryCatchInTearDown()
+        {
+            string expectedButtonText = "Invalid button text for test failure";
+
+            loginPage.Login(Configurator.ReadConfiguration().UserNameSauceDemo, Configurator.ReadConfiguration().PasswordSauceDemo);
+            logger.Debug("Successful login");
+
+            productsPage.ProductButton("Sauce Labs Fleece Jacket").Click();
+            logger.Debug("Clicked on product button");
+
+            string actualButtonText = productsPage.ProductButton("Sauce Labs Fleece Jacket").Text;
+            logger.Debug("Got button text");

```

Comment:
Also you can send button text into logger to see from logs which text you got
example: logger.Debug($"Got button text: {actualButtonText}" );